### DISCUSSION TITLE
Is there any easy Uno way to blur some controls or text

### DISCUSSION THREAD

**Discussion by c-npolyak** [unoplatform/uno#11621] [General]
Friends, we are using UNO for Wasm and UWP. Is there any easy Uno way to blur some controls or text? We are now trying SkiaSharp, applying SKXamlCanvas to every visible text block within a data grid? Are there any better ways? Is there a way perhaps to turn the visible part of a data grid into an image and then blur the whole image instead of every text block? BTW we tried Acrylic brush and it was not working well for our purpose. Any help is greatly appreciated.

**Comment by jeromelaban** [comment]
Can you give a visual example of what you mean by "blurring controls or text"?

**Reply by jeromelaban** [reply]
_Make sure to continue in the same thread_

Thanks. Can you still provide a visual example of what you mean?

**Reply by c-npolyak** [reply]
![image](https://user-images.githubusercontent.com/127349248/223861251-aad7ddd4-a228-433d-b80d-f62cf1269dd0.png)

**Reply by c-npolyak** [reply]
above is an example of blurring a grid

Here is another example of blurring out a panel (on the right):
![image](https://user-images.githubusercontent.com/127349248/223861534-bfdae665-ade0-42d6-878f-84eff500505e.png)

**Reply by jeromelaban** [reply]
Got it, thanks. The acrylic brush is present to do exactly this: https://playground.platform.uno/#fe947ab3.

What is not working for you with this brush ?

**Comment by c-npolyak** [comment]
Ideally I want to blur any part of the Visual Tree starting from some element and down. If there is no such functionality - at least I want to be able to blur the leaf parts of the visual tree, e.g. text and icons.

**Reply by FrozDark** [reply]
To blur something with acrylic brush you have to overlay those controls with something like Grid with background set to acrylic brush

**Reply by c-npolyak** [reply]
In my experience Acrylic brush is ok to blur large images, but not small text. It blurs it beyond what our clients expect.

**Reply by jeromelaban** [reply]
At this point, on the Uno APIs do not provide control over the blurring effect. 

Still it's possible to create a [custom control for Wasm specifically](https://platform.uno/docs/articles/interop/wasm-javascript-1.html?q=javascript), which uses the `blur()` CSS API to create a custom overlay. It will be a platform specific implementation and if you need other platforms, you'll need to create other controls for those platforms.

**Reply by c-npolyak** [reply]
Thank Jerome!
